community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/ExplicitIndexProxy.java

```diff
@@ -27,211 +28,224 @@
+        {
+            operations.nodeExplicitIndexDrop( name );
+        }
+
+        @Override
+        public long id( PropertyContainer entity )
+        {
+            return ((Node) entity).getId();
+        }
+    };
+
+    public static final Type RELATIONSHIP = new Type<Relationship>()
+    {
+        @Override
+        public Class<Relationship> getEntityType()
+        {
+            return Relationship.class;
+        }
+
+        @Override
+        public Relationship entity( long id, GraphDatabaseService graphDatabaseService )
+        {
+            return graphDatabaseService.getRelationshipById( id );
+        }
+
+        @Override
+        public IndexHits<Relationship> get( KernelTransaction ktx, String name, String key, Object value,
+            GraphDatabaseService graphDatabaseService ) throws ExplicitIndexNotFoundKernelException
+        {
+            RelationshipExplicitIndexCursor cursor = ktx.cursors().allocateRelationshipExplicitIndexCursor();
+            ktx.indexRead().relationshipExplicitIndexLookup( cursor, name, key, Values.of( value ), -1, -1 );
+            return new CursorWrappingRelationshipIndexHits( cursor, graphDatabaseService );
+        }
+
+        @Override
+        public IndexHits<Relationship> query( KernelTransaction ktx, String name, String key, Object queryOrQueryObject,
+            GraphDatabaseService graphDatabaseService )
+                throws ExplicitIndexNotFoundKernelException
+        {
+            RelationshipExplicitIndexCursor cursor = ktx.cursors().allocateRelationshipExplicitIndexCursor();
+            ktx.indexRead().relationshipExplicitIndexQuery( cursor, name, key, queryOrQueryObject,-1, -1 );
+            return new CursorWrappingRelationshipIndexHits( cursor, graphDatabaseService );
+        }
+
+        @Override
+        public IndexHits<Relationship> query( KernelTransaction ktx, String name, Object queryOrQueryObject,
+            GraphDatabaseService graphDatabaseService )
+                throws ExplicitIndexNotFoundKernelException
+        {
+            RelationshipExplicitIndexCursor cursor = ktx.cursors().allocateRelationshipExplicitIndexCursor();
+            ktx.indexRead().relationshipExplicitIndexQuery( cursor, name, queryOrQueryObject, -1 , -1 );
+            return new CursorWrappingRelationshipIndexHits( cursor, graphDatabaseService );
+        }
+
+        @Override
+        public void add( ExplicitIndexWrite operations, String name, long id, String key, Object value )
+                throws ExplicitIndexNotFoundKernelException, EntityNotFoundException
+        {
+            operations.relationshipAddToExplicitIndex( name, id, key, value );
+        }
+
+        @Override
+        public void remove( ExplicitIndexWrite operations, String name, long id, String key, Object value )
+                throws ExplicitIndexNotFoundKernelException
+        {
+            operations.nodeRemoveFromExplicitIndex( name, id, key, value );
```

Comment:
From here to the end of `RELATIONSHIP` there seems to be some copy-paste things which are still `node` instead of `relationship`

Reply:
I think that is fixed in a later commit?

Reply:
Or at this line it should really be node but at one point there was an error further down in the file where we remove relationships.

Reply:
Yes, it's already fixed. Carry on =).